string_match_without_scan_or_regex.rb

```diff
@@ -1,36 +1,63 @@
-def match?(string, substring)
-  string_array = string.split("")
-  substring_array = substring.split("")
-  character_matches = []
-  substring_array.each_with_index do |substring_character, substring_index|
-    find_character_match(string_array, substring_array, substring_character, substring_index, character_matches)
+class String
```

Comment:
extending core classes is almost always a bad idea.  I know it seems super cool when you first come to Ruby, but read around about how "monkey-patching" can go horribly wrong.